src/drt/src/db/drObj/drAccessPattern.cpp

```diff
@@ -0,0 +1,105 @@
+/* Authors: Osama */
+/*
+ * Copyright (c) 2022, The Regents of the University of California
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the University nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE REGENTS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "db/drObj/drAccessPattern.h"
+
+#include "distributed/frArchive.h"
+using namespace fr;
+template <class Archive>
+void drAccessPattern::serialize(Archive& ar, const unsigned int version)
+{
+  (ar) & boost::serialization::base_object<drBlockObject>(*this);
+  (ar) & beginPoint_;
+  (ar) & beginLayerNum_;
+  (ar) & beginArea_;
+  (ar) & mazeIdx_;
+  (ar) & pin_;
+  (ar) & validAccess_;
+  (ar) & vUIdx_;
+  (ar) & vDIdx_;
+  (ar) & onTrackX_;
+  (ar) & onTrackY_;
+  (ar) & pinCost_;
+  // vU_ and vD_ are pointers to vectors based on previous implementation of dr.
+  // Each read serialization, new vU_ and vD_ are created and kept in heap
+  // without deletion.
+  // TODO: fix memory leaked here.
```

Comment:
I'm not sure. There's definitely memory leaked here because the pointer to the vector does not get deleted. However it does not affect the current test cases.

Reply:
ok then give yourself a jira to follow up on it.

Reply:
https://openroad.atlassian.net/browse/OR-1331